overlord/snapstate/snapmgr.go

```diff
@@ -560,6 +660,12 @@ func (m *SnapManager) Ensure() error {
 
 	m.runner.Ensure()
 
+	// needs to run after the runner
+	if err := m.ensureSnapConfineApparmor(); err != nil {
+		// FIMXE: this will spam the world as ensure runs often
```

Comment:
We have some options here, I'm happy to rework it. The implementation is done like this because:
a) on startup we need to check if we have an apparmor profile for snap-confine. we may upgrade/re-exec from an older version of snapd so we need to check this, the ensure loop seems to be a good place for this
b) if we need to have this code anyway it can be reused for the case when the core updates. please note that it will not actually re-generate profiles all the time. if the profile exists, it will just do nothing.

We can of course change that to:
a) check *once* on startup if the profile is there
b) write the snap-confine profile in e.g. the setup-security task of core (as a special case there).

Happy to rework if you think the later design is cleaner (or anyother design that is cleaner!).